packages/components/src/components/post-pagination/post-pagination.tsx

```diff
@@ -0,0 +1,844 @@
+    if (newWidth === this.lastWindowWidth) {
+      return;
+    }
+    
+    this.lastWindowWidth = newWidth;
+    this.measureAndCalculateVisiblePages();
+  }
+
+  /**
+   * Calculates gap between two elements
+   */
+  private calculateGap(first: Element, second: Element): number {
+    const firstRect = first.getBoundingClientRect();
+    const secondRect = second.getBoundingClientRect();
+    return secondRect.left - firstRect.right;
+  }
+
+  /**
+   * Measures actual rendered elements to determine how many pages can fit
+   */
+  private measureAndCalculateVisiblePages() {
+    if (!this.navRef || !this.hiddenItemsRef || !this.connected) {
+      return;
+    }
+
+    const totalPages = this.getTotalPages();
+    if (totalPages <= 1) {
+      return;
+    }
+
+    const paginationPadding = this.getPaginationPadding();
+    const availableWidth = this.getAvailableWidth();
+    const netWidth = availableWidth - paginationPadding;
+
+    const controlButtonsWidth = this.getControlButtonsWidth();
+    const pageButton = this.hiddenItemsRef.querySelector('.hidden-page-button');
+    const ellipsis = this.hiddenItemsRef.querySelector('.hidden-ellipsis');
+    
+    if (!pageButton) {
+      return;
+    }
+
+    const singleButtonWidth = pageButton.getBoundingClientRect().width;
+    const gap = pageButton && ellipsis ? this.calculateGap(pageButton, ellipsis) : 0;
+    const widthForPages = netWidth - controlButtonsWidth;
+
+    // Calculate how many page buttons can fit
+    const maxPages = Math.floor((widthForPages + gap) / (singleButtonWidth + gap));
+    const clampedMaxPages = Math.max(MIN_VISIBLE_PAGES, Math.min(maxPages, totalPages));
+    
+    this.maxVisiblePages = clampedMaxPages;
+    this.updatePagesWithValidation();
+  }
+
+  /**
+   * Gets total width of control buttons (prev/next)
+   */
+  private getControlButtonsWidth(): number {
+    if (!this.hiddenItemsRef) return 0;
+    
+    const controlButtons = Array.from(
+      this.hiddenItemsRef.querySelectorAll('.hidden-control-button')
+    );
+    
+    const totalWidth = controlButtons.reduce((sum, el) => {
+      return sum + el.getBoundingClientRect().width;
+    }, 0);
+    
+    return totalWidth;
+  }
+
+  /** 
+   * Gets the horizontal padding of the pagination container
+   */
+  private getPaginationPadding(): number {
+    if (!this.navRef) return 0;
+    const computedStyle = window.getComputedStyle(this.navRef);
+    const paddingLeft = parseFloat(computedStyle.paddingLeft) || 0;
+    const paddingRight = parseFloat(computedStyle.paddingRight) || 0;
+    return paddingLeft + paddingRight;
+  }
+
+  /**
+   * Gets available width from parent container
+   */
+  private getAvailableWidth(): number {
+    if (!this.navRef) return 0;
+    
+    let parent = this.host.parentNode;
+    while (parent && !(parent instanceof HTMLElement)) {
+      parent = parent.parentNode;
+    }
+    
+    return parent instanceof HTMLElement ? parent.clientWidth : window.innerWidth;
```

Comment:
Check if you can use `parentElement` instead of `parentNode` and get rid of all the typing noise.

But thinking about it again, don't you end up at the `html` element always anyway? And if you do so, the next while loop will set `parent = html.parentNode` which returns `null`. Then below, you always would get the `window.innerWidth` as a return value and the whole function becomes obsolete.

Please check that as well.